### PR TITLE
Robust Integration Tests

### PR DESCRIPTION
**Pull request by wistuba**
Tests now work reliably on Mac and Github Ubuntu systems.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/integration_tests/configs/suites/quick/avalanche-er.json [hunk 1]
```diff
@@ -5,5 +5,6 @@
   "dataset": "cifar10.json",
   "backend": "local",
   "job_name": "class-incremental-mlp-avalanche-er",
-  "expected_accuracy": [0.14100000262260437, 0.6794999837875366]
+  "expected_accuracy_linux": [[0.14100000262260437, 0.6794999837875366]],
```
Comment (prabhuteja12) [inline test/integration_tests/configs/suites/quick/avalanche-er.json:8]: is a 4% difference in performance attributable to platforms?

Comment (wistuba) [reply test/integration_tests/configs/suites/quick/avalanche-er.json:8]: the gap should close if we increase number of epochs from 3 to 50

File: .github/workflows/run_renate.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ env:
 
 jobs:
   run:
-    runs-on: Renate_ubuntu-latest_16-core
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (wistuba) [reply .github/workflows/run_renate.yml:33]: not sure how many cores ubuntu-latest has but I didn't experience a big difference in time. an earlier PR also required 6min for avalanche-er. could be, that we are currently not making good use of these additional resources or the model is too small to allow for it.

File: test/integration_tests/configs/suites/quick/er.json [hunk 3]
```diff
@@ -5,5 +5,6 @@
   "dataset": "cifar10.json",
   "backend": "local",
   "job_name": "class-incremental-mlp-er",
-  "expected_accuracy": [0.5799999833106995, 0.367000013589859]
+  "expected_accuracy_linux": [[0.5799999833106995, 0.367000013589859], [0.35179999470710754, 0.3208000063896179]],
```
[the remark this answers is not in the document]
Comment (wistuba) [reply test/integration_tests/configs/suites/quick/er.json:8]: it appears as if we may see 2 different results in github (possibly because of hardware). the test will pass if we see any of these results

File: src/renate/benchmark/experimentation.py [hunk 4]
```diff
@@ -252,7 +252,7 @@ def _execute_experiment_job_locally(
     See renate.benchmark.experimentation.execute_experiment_job for more details.
     """
     logger.info("Start experiment.")
-    seed_everything(seed)
+    seed_everything(seed, True)
```
[the remark this answers is not in the document]
Comment (wistuba) [reply src/renate/benchmark/experimentation.py:255]: not super sure but according to my interpretation of https://pytorch-lightning.readthedocs.io/en/1.6.5/api/pytorch_lightning.utilities.seed.html#pytorch_lightning.utilities.seed.seed_everything it will seed the workers

[the remark this answers is not in the document]
Comment (wistuba) [reply src/renate/benchmark/experimentation.py:255]: https://pytorch.org/docs/stable/notes/randomness.html#dataloader

File: test/integration_tests/configs/suites/quick/er.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (prabhuteja12) [reply test/integration_tests/configs/suites/quick/er.json:8]: I meant that taking element [0][0], [1][0] in linux and [0][0] in darwin seem extremely different.

**Review by prabhuteja12 — DISMISSED** [review]
assuming that the performance variance is largely attributable to platform issues, the rest seems fine to me.

**Review by prabhuteja12 — APPROVED** [review]
